src/store/modules/job/actions.ts

```diff
@@ -765,6 +766,126 @@ const actions: ActionTree<JobState, RootState> = {
+    commit(types.JOB_ADDED_TO_BULK, payload);
+  },
+  async scheduleBulkJobs({ dispatch }, payload) {
+    // TODO: handle case for bulk jobs which are already scheduled. 
+    // As of now, the job will get scheduled again even if it is pending
+    const jobParams = [] as any;
+    let failedJobs = 0;
+    payload.shopifyConfigs.map((shopId: string) => {
+      return payload.jobs.map((job: any) => {
+        const params = {
+          'JOB_NAME': job.jobName,
+          'SERVICE_NAME': job.serviceName,
+          'SERVICE_COUNT': '0',
+          'SERVICE_TIME': job.setTime.toString(),
+          'SERVICE_TEMP_EXPR': job.frequency,
+          'SERVICE_RUN_AS_SYSTEM': 'Y',
+          'jobFields': {
+            'productStoreId': payload.eComStoreId,
+            'systemJobEnumId': job.systemJobEnumId,
+            'tempExprId': job.frequency, // Need to remove this as we are passing frequency in SERVICE_TEMP_EXPR, currently kept it for backward compatibility
+            'maxRecurrenceCount': '-1',
+            'parentJobId': job.parentJobId,
+            'runAsUser': 'system', //default system, but empty in run now.  TODO Need to remove this as we are using SERVICE_RUN_AS_SYSTEM, currently kept it for backward compatibility
+            'recurrenceTimeZone': this.state.user.current.userTimeZone
+          },
+          'statusId': "SERVICE_PENDING",
+          'systemJobEnumId': job.systemJobEnumId
+        } as any
+
+        
+        if (job?.runtimeData?.shopifyConfigId) {
+          const shopifyConfig = this.state.user.shopifyConfigs.find((config: any) => {
+            return config.shopId === shopId;
+          })
+
+          params['shopifyConfigId'] = shopifyConfig.shopifyConfigId;
+          params['jobFields']['shopId'] = shopId;
+        }
+
+        // checking if the runtimeData has productStoreId, and if present then adding it on root level
+        job?.runtimeData?.productStoreId?.length >= 0 && (params['productStoreId'] = payload.eComStoreId)
+        job?.priority && (params['SERVICE_PRIORITY'] = job.priority.toString())
+
+        // assigning '' (empty string) to all the runtimeData properties whose value is "null"
+        job.runtimeData && Object.keys(job.runtimeData).map((key: any) => {
+          if (job.runtimeData[key] === 'null') job.runtimeData[key] = ''
+        })
+        jobParams.push({ ...job.runtimeData, ...params, jobId: job.jobId });
+      })
+    })
+
+    Promise.allSettled(jobParams.map(async (param: any) => {
+      const resp: any = await JobService.scheduleJob(param);
+      if(resp.status === 200 && !hasError(resp)){
+        // Removing the scheduled job
+        dispatch('removeBulkJob', param.jobId);
+        return Promise.resolve(resp);
+      } else {
+        failedJobs++;
+        return Promise.reject(resp);
+      }
+    })).then((resps: any) => {
+      resps.some((resp: any) => {
```

Comment:
I think this might fail if the request other than first one fails

Reply:
Yes, sir, updated the code to handle the case when jobs other than the first one fail.